### PR TITLE
Migration/barnote

### PR DESCRIPTION
**Pull request by rvilarl**
BarNote & BarLine!

### CODE REVIEW

File: src/barnote.ts [hunk 1]
```diff
@@ -43,34 +53,31 @@ export class BarNote extends Note {
   }
 
   // Get and set the type of Bar note. `type` must be one of `Vex.Flow.Barline.type`.
-  getType() {
+  getType(): number {
```
Comment (mscuthbert) [inline src/barnote.ts:56]: Can this import the enum from stavebarline and classify that as the type?

File: src/barnote.ts [hunk 2]
```diff
@@ -43,34 +53,31 @@ export class BarNote extends Note {
...
     return this.type;
   }
-  setType(type) {
+
+  setType(type: string | number): this {
```
Comment (mscuthbert) [inline src/barnote.ts:60]: similarly instead of number can we put BarlineType?

File: src/stavebarline.ts [hunk 3]
```diff
@@ -3,25 +3,35 @@
...
-      REPEAT_BOTH: 6,
-      NONE: 7,
-    };
+
+  static get type(): typeof BarlineType {
+    return BarlineType;
   }
 
-  static get typeString() {
+  static get typeString(): Record<string, number> {
```
Comment (mscuthbert) [inline src/stavebarline.ts:34]: can we narrow the Record with literals so it's `Record<BarlineTypeString, BarlineType>`?

File: src/stavebarline.ts [hunk 4]
```diff
@@ -36,7 +46,7 @@ export class Barline extends StaveModifier {
   /**
    * @constructor
    */
-  constructor(type) {
+  constructor(type: number | string) {
```
Comment (mscuthbert) [inline src/stavebarline.ts:49]: As above.

**Review by mscuthbert — COMMENTED** [review]
Looks extremely good -- I was wondering if we can specify the string and number of types more closely in the typing?  Especially in barnote.

File: src/stavebarline.ts (hunk 3 above)
Comment (rvilarl) [reply src/stavebarline.ts:34]: How would you declare `BarlineTypeString` if we make `enum` we change the interface. 
`BarlineType` change done.

File: src/barnote.ts (hunk 2 above)
Comment (rvilarl) [reply src/barnote.ts:60]: done

File: src/barnote.ts (hunk 1 above)
Comment (rvilarl) [reply src/barnote.ts:56]: done

File: src/stavebarline.ts (hunk 3 above)
Comment (rvilarl) [reply src/stavebarline.ts]: @0xfe any suggestion on how to declare `BarlineTypeString`? If `string` is ok, this is ready to merge! :)

**Comment by rvilarl** [comment]
#609 changes added and tested.

**Comment by rvilarl** [comment]
@0xfe anything pending here?

**Comment by 0xfe** [comment]
Thanks @rvilarl. Can you attach a screenshot of the test here please?

**Comment by rvilarl** [comment]
image

**Comment by rvilarl** [comment]
@0xfe is that right?

File: src/barnote.ts [hunk 5]
```diff
@@ -10,14 +10,24 @@
 
 import { Vex } from './vex';
 import { Note } from './note';
-import { Barline } from './stavebarline';
+import { Barline, BarlineType } from './stavebarline';
+
+export interface BarnoteMetrics {
```
Comment (0xfe) [inline src/barnote.ts:15]: Let's inline this please.

**Review by 0xfe — COMMENTED** [review]
Test looks perfect. Minor comment.

File: src/barnote.ts (hunk 5 above)
Comment (rvilarl) [reply src/barnote.ts]: done

**Comment by rvilarl** [comment]
@0xfe Ready to merge!

**Comment by 0xfe** [comment]
Great, merging!

**Pull request merged**